docs/src/main/asciidoc/security-oidc-code-flow-authentication-concept.adoc

```diff
@@ -267,285 +660,167 @@ quarkus.oidc.logout.post-logout-uri-param=returnTo
 
-link:https://datatracker.ietf.org/doc/html/rfc7636[Proof Key for Code Exchange] (PKCE) minimizes the risk of the authorization code interception.
+[[front-channel-logout]]
+===== Front-channel logout
 
-While `PKCE` is of primary importance to the public OpenID Connect clients (such as the SPA scripts running in a browser), it can also provide an extra level of protection to Quarkus OIDC `web-app` applications which are confidential OpenID Connect clients capable of securely storing the client secret and using it to exchange the code for the tokens.
+link:https://openid.net/specs/openid-connect-frontchannel-1_0.html[Front-Channel Logout] can be used to logout the current user directly from the user agent (e.g. its browser).
+It is like <<back-channel-logout,Back-channel logout>> but the logout steps are executed by the user agent like the browser and not in the background by the OIDP provider.
+This option is rarely used.
 
-You can enable `PKCE` for your OIDC `web-app` endpoint with a `quarkus.oidc.authentication.pkce-required` property and a 32 characters long secret, for example:
+You can configure Quarkus to support `Front-Channel Logout` as follows:
 
-[source, properties]
+[source,properties]
 ----
-quarkus.oidc.authentication.pkce-required=true
-quarkus.oidc.authentication.pkce-secret=eUk1p7UB3nFiXZGUXi0uph1Y9p34YhBU
+quarkus.oidc.auth-server-url=http://localhost:8180/realms/quarkus
+quarkus.oidc.client-id=frontend
+quarkus.oidc.credentials.secret=secret
+quarkus.oidc.application-type=web-app
+
+quarkus.oidc.logout.frontchannel.path=/front-channel-logout
 ----
 
-If you already have a 32 characters long client secret then `quarkus.oidc.authentication.pkce-secret` does not have to be set unless you prefer to use a different secret key.
+This path will be compared against the current request's path and the user will be logged out if these paths match.
 
-The secret key is required for encrypting a randomly generated `PKCE` `code_verifier` while the user is being redirected with the `code_challenge` query parameter to OpenID Connect Provider to authenticate. The `code_verifier` will be decrypted when the user is redirected back to Quarkus and sent to the token endpoint alongside the `code`, client secret and other parameters to complete the code exchange. The provider will fail the code exchange if a `SHA256` digest of the `code_verifier` does not match the `code_challenge` provided during the authentication request.
+[[local-logout]]
+===== Local logout
 
-=== Listening to important authentication events
+<<user-initiated-logout,User-initiated logout>> will log the person out from the OIDC provider, if it is used as single sign on, it might not be what you desire.
```

Comment:
```suggestion
<<user-initiated-logout,User-initiated logout>> will log the user out of the OIDC provider, if it is used as single-sign on, it might not be what you require.
```